pkg/kv/kvprober/kvprober.go

```diff
@@ -425,7 +425,7 @@ func (p *Prober) readProbeImpl(ctx context.Context, ops proberOpsI, txns proberT
 
 	d := timeutil.Since(start)
 	// Extract leaseholder information from the trace recording if enabled.
-	if tracingEnabled.Get(&p.settings.SV) {
+	if tracingEnabled.Get(&p.settings.SV) && finishAndGetRecording != nil {
```

Comment:
Why is `finishAndGetRecording` null? Is it bc/ tracing was enabled part way thru the trace? If yes, I think a clearer way to write the bug fix is to check the cluster setting a single time at the top of this function (store it in a local variable for use later), instead of checking it twice.

Reply:
I looked at the occurrences of the error in sentry and this happened in 2 separate clusters only once, which matches with the flag changing inbetween the 2 calls.
<img width="345" alt="signal_unix_go_881__sigpanic___runtime_errorString__runtime_error__invalid_memory_address_or_nil_pointer_dereference_via__withstack_withStack_—_cockroach-labs_—_cockroachdb" src="https://github.com/user-attachments/assets/fc92ac3f-4615-400e-943d-9e8c6891602f">

Taking the cluster id `b1c618ed-a9ff-4264-b623-dff46c7c90ae` which is `crl-staging-8k4c`, the crash happened at `Oct 1, 2024 4:15:23 PM UTC` which corresponds to when the flag was switched [ddog log](https://us5.datadoghq.com/logs?query=crl-staging-8k4c%20set%20kv.prober.tracing.enabled%20from%20false%20to%20true%20&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=paused&storage=flex_tier&stream_sort=time%2Casc&viz=stream&from_ts=1727799237029&to_ts=1727799608353&live=false)